collectors/prometheus.go

```diff
@@ -150,8 +150,10 @@ func (p *PrometheusExporter) Start() error {
 
 	// Start the htlc monitor goroutine. This will subscribe to htlcs and
 	// update all of our routing-related metrics.
-	if err := p.htlcMonitor.start(); err != nil {
-		return err
+	if !p.monitoringCfg.DisableHtlc {
```

Comment:
It probably won't fail, but just for completeness' sake we should also surround the `stop` with the same if.

Reply:
Updated